src/annotation_operators.ts

```diff
@@ -238,56 +237,144 @@ function get_distance_from_point_to_line(point_annotation: ULabelAnnotation, lin
-                distance_from[line_class_id] = distance
+/**
+ * Assigns a single point annotation a distance to the closest polyline
+ * from each diffrent class of polyline, as well as the distance from
+ * the closest line, regardless of class.
+ * 
+ * @param point_annotation single point annotation
+ * @param line_annotations set of line annotations
+ * @param offset Offset of a particular annotation in the set. Used when an annotation is being moved by the user
+ */
+export function assign_closest_line_to_single_point(
+    point_annotation: ULabelAnnotation,
+    line_annotations: ULabelAnnotation[],
+    offset: Offset = null
+) {
+    // Create a new distance_from object for the point annotation
+    let distance_from: DistanceFromPolylineClasses = {closest_row: undefined}
+
+    // Calculate the distance from each line and populate the distance_from accordingly
+    line_annotations.forEach(current_line => {
+        const line_class_id = get_annotation_class_id(current_line)
+        const distance = get_distance_from_point_to_line(point_annotation, current_line, offset)
+
+        // If the distance from the current class is undefined, then set it
+        // Otherwise replace the value if the current distance is less than the one set
+        if (distance_from[line_class_id] === undefined || distance < distance_from[line_class_id].distance) {
+            distance_from[line_class_id] = {
+                distance: distance, 
+                polyline_id: current_line.id,
             }
+        }
 
-            // Likewise check to see if the current distance is less than a line of any class
-            if (distance_from["single"] === undefined || distance < distance_from["single"]) {
-                distance_from["single"] = distance
+        // Likewise check to see if the current distance is less than a line of any class
+        if (distance_from.closest_row === undefined || distance < distance_from.closest_row.distance) {
+            distance_from.closest_row = {
+                distance: distance,
+                polyline_id: current_line.id,
             }
-        })
+        }
+    })
 
-        // Assign the distance from object to the current point
-        current_point.distance_from = distance_from
+    // Assign the distance from object to the current point
+    point_annotation.distance_from = distance_from
+}
+
+
+/**
+ * Update the distance from a single line to each point annotation.
+ * If a line was previously the closest line to a point, then the distance from that point to ALL lines will be recalculated.
+ * Used when a single polyline is modified.
+ * 
+ * @param line_annotation line annotation
+ * @param point_annotations all point annotations
+ * @param all_line_annotations all line annotations
+ * @param offset offset of the line annotation
+ */
+export function update_distance_from_line_to_each_point(
```

Comment:
When a line is updated, we don't have to update *all* the distances from every point to every line. Instead:
- For points that had the line as one of its saved ids in its `distance_from`, we recalculate that point's distance to EVERY line, since the current line may have moved farther away.
- For points that do NOT have the line as one of its saved ids, we can simply check if the current line's distance to the point is smaller than that of any of the point's saved distances.